lib/validator.js

```diff
@@ -39,7 +40,9 @@ async function validateExpectations() {
 		}
 	}
 
-	if (req.size !== exp.size) {
+	let expSize = exp.size;
+	if (exp.has("rpId")) expSize--; // don't include optional fields
```

Comment:
This feels like we should find a more beautiful solution, eg making a list of optional fields (referencing it in the documentation) and count all non-optional fields, what do you think? This way we can more easily extend this later.

Reply:
agree with you, it would be more clean and maintainable, doing it now